src/casp/services/api/exception_handlers.py

```diff
@@ -1,8 +1,22 @@
 from fastapi import Request, responses, status
+from starlette_context import context
 
 from casp.services._auth.exceptions import TokenException
 from casp.services.api.data_manager.exceptions import NotFound
 from casp.services.api.services.exceptions import APIBaseException
+from casp.services.constants import ContextKeys, HeaderKeys
+
+
+def decorate_response_headers(init_headers: dict[str, str] = None):
```

Comment:
I think we should eventually start using "list", "dict", instead of "t.Dict", "t.List" as they got outdated after 3.10. But since we use it everywhere, shouldn't we keep doing them until we change them all at once? If not, then maybe I should start using newer annotations as well?

Reply:
Maybe I'll keep this and do a followon to convert all instances to the more native `list` and `dict` if that's ok